runtime/src/in_mem_accounts_index.rs

```diff
@@ -26,8 +26,8 @@ impl<V: IsCached> InMemAccountsIndex<V> {
         self.map.entry(pubkey)
     }
 
-    pub fn iter(&self) -> Iter<K, AccountMapEntry<V>> {
-        self.map.iter()
+    pub fn iter(&self) -> Vec<(K, AccountMapEntry<V>)> {
+        self.map.iter().map(|(k, v)| (*k, v.clone())).collect()
```

Comment:
I understand the intent here, but I do find it odd to have a function called `iter()` that _doesn't_ return an iterator. I'll see if I can poke around other rust stuffs to see if there's already a name for this. Maybe something like `to_vec()` would work?